tests/lib/mirrors/reflected_type_generics_test.dart

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) 2013, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+library test.reflected_type_generics_test;
+
+import 'dart:mirrors';
+
+import 'package:expect/expect.dart';
+
+import 'reflected_type_helper.dart';
+
+class A<T> {}
+class P {}
+class B extends A<P> {}
+class C<K, V> {}
+class D<T> extends A<T> {}
+class E<K> extends C<K, int> {}
+
+main() {
+  // "Happy" paths:
+  expectReflectedType(reflectType(A, [P]), new A<P>().runtimeType);
+  expectReflectedType(reflectType(C, [B, P]), new C<B, P>().runtimeType);
+  expectReflectedType(reflectType(D, [P]), new D<P>().runtimeType);
+  expectReflectedType(reflectType(E, [P]), new E<P>().runtimeType);
+
+  // Edge cases:
+  Expect.throws(
+    () => reflectType(P, []),
+    (e) => e is ArgumentError && e.invalidValue is List,
+    "Should throw an ArgumentError if reflecting not a generic class with empty list of type arguments");
```

Comment:
As written above: I wouldn't test that it really returns an argument error.

Since you already wrote the tests, keep them.

Long lines: 31, and 35.